ruby_event_store-rom/lib/ruby_event_store/rom/repositories/events.rb

```diff
@@ -51,6 +51,18 @@ def read(direction, stream, from:, limit:, batch_size:)
                 .to_ary
             end
             BatchEnumerator.new(batch_size, limit || Float::INFINITY, reader).each
+          elsif read_as == RubyEventStore::Specification::FIRST
+            stream_entries
+              .ordered(direction, stream, offset_entry_id)
+              .combine(:event)
+              .map_with(:stream_entry_to_serialized_record, auto_struct: false)
+              .to_ary.first
```

Comment:
@mpraglowski I think you don't want `.to_ary` here -- just `first` should get you one record

Reply:
right, `first` is implemented here